release/models/nat/openconfig-nat.yang

```diff
@@ -0,0 +1,1008 @@
+        "Counters for NAT translation entry operations.";
+
+      leaf packet-count-inbound {
+        type yang:counter64;
+        description
+          "Number of inbound packets (external to internal) for
+          this translation.";
+      }
+
+      leaf packet-count-outbound {
+        type yang:counter64;
+        description
+          "Number of outbound packets (internal to external) for
+          this translation.";
+      }
+
+      leaf byte-count-inbound {
+        type yang:counter64;
+        description
+          "Number of inbound bytes (external to internal) for
+          this translation.";
+      }
+
+      leaf byte-count-outbound {
+        type yang:counter64;
+        description
+          "Number of outbound bytes (internal to external) for
+          this translation.";
+      }
+    }
+  }
+
+  grouping nat-translation-table-top {
+    description
+      "Top-level grouping for NAT translation table.";
+
+    container translations {
+      config false;
+      description
+        "Active NAT translation entries showing current
+        address/port mappings.";
+
+      list translation {
+        key "translation-id";
+        description
+          "List of active NAT translation entries.";
+
+        leaf translation-id {
+          type leafref {
+            path "../state/translation-id";
+          }
+          description
+            "Reference to translation entry identifier.";
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for the NAT translation entry.";
+
+          leaf translation-id {
+            type uint64;
+            description
+              "Unique identifier for the translation entry.";
+          }
+
+          uses nat-translation-entry-state;
+        }
+      }
+    }
+  }
+
+  grouping nat-instance-top {
+    description
+      "Top-level grouping for NAT instances.";
+
+    container instances {
+      description
+        "NAT instance configuration and state.";
+
+      list instance {
+        key "name";
+        description
+          "List of NAT instances.";
+
+        leaf name {
+          type leafref {
+            path "../config/name";
+          }
+          description
+            "Reference to NAT instance name.";
+        }
+
+        container config {
+          description
+            "Configuration parameters for the NAT instance.";
+
+          uses nat-instance-config;
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for the NAT instance.";
+
+          uses nat-instance-config;
+          uses nat-instance-state;
+        }
+
+        uses nat-interface-top;
+        uses nat-pool-top;
+        uses nat-mapping-top;
+        uses nat-translation-table-top;
+      }
+    }
+  }
+
+  grouping nat-pool-top {
+    description
+      "Top-level grouping for NAT pools.";
+
+    container pools {
```

Comment:
instead of "pools" for dynamic NAT and "mappings" for static NAT (which I find to be ambiguous), can we use the terms "dynamic" and "static" which seems to match the terminology used across vendors?